wifi-scripts: change wifi-station's mac option into list

rany2 opened the pull request:
In the past PR[1] to add SAE wifi-station support, a commenter[2] requested that the mac option be changed into a list. After trying to migrate my old RADIUS setup I found myself wanting this change as well as it would simplify my config. This patch does precisely that. Old configs that specify `option mac ....` still work without any issues.

This change was done for both PSK and SAE. The schema was updated as well.

[1]: https://github.com/openwrt/openwrt/pull/17145
[2]: https://github.com/openwrt/openwrt/pull/17145#issuecomment-2523507953

b0ch3nski commented:
This is indeed a right way to go - makes mac based filtering much easier :+1:

Neustradamus commented:
@rany2: Nice PR!

geldot commented on package/network/config/wifi-scripts/files-ucode/usr/share/schema/wireless.wifi-station.json:
```diff
@@ -6,7 +6,10 @@
 	"properties": {
 		"mac": {
 			"description": "The stations MAC",
```
Can we also fix the description while we're here? `"The station's MAC addresses"`

rany2 replied:
Done :)

blogic commented:
it looks like the schema for the ucode version of the scripts was changed but the actual ucode script was not changed.

rany2 commented:
> it looks like the schema for the ucode version of the scripts was changed but the actual ucode script was not changed.

At the time I wasn't to find any code to change. Can you point me to where it's used?

blogic commented:
https://github.com/openwrt/openwrt/blob/main/package/network/config/wifi-scripts/files-ucode/usr/share/ucode/wifi/ap.uc#L297

rany2 commented:
@blogic thanks, do I need to also modify it for SAE support? I wasn't aware of it.

rany2 commented:
@blogic I'm not sure if this is correct currently. Please let me know

rany2 commented:
I'll make a separate PR to add SAE wifi-station support for the ucode implementation.

Neustradamus commented:
@rany2: Have you progressed on your PR?

rany2 commented:
@Neustradamus Clearly not, I forgot about it. I'll make the time to fix it up on the weekends maybe

rany2 commented:
> Is there anything else that needs to be done to get this merged? :shipit:

That referenced PR still hasn't been merged, so until then I guess we need to wait.

rany2 committed:
wifi-scripts: change wifi-station's mac option into list

In the past PR[1] to add SAE wifi-station support, a commenter[2] requested
that the mac option be changed into a list. After trying to migrate my old
RADIUS setup I found myself wanting this change as well as it would simplify
my config. This patch does precisely that. Old configs that specify
`option mac ....` still work without any issues.

This change was done for both PSK and SAE. The schema was updated as well.

[1]: https://github.com/openwrt/openwrt/pull/17145
[2]: https://github.com/openwrt/openwrt/pull/17145#issuecomment-2523507953

Signed-off-by: Rany Hany <rany_hany@riseup.net>
Link: https://github.com/openwrt/openwrt/pull/17650
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

Neustradamus commented:
I hope that @nbd168, @blogic, @robimarko, @hauke can look your PR after your update :)

Neustradamus commented:
@hauke: Have you seen this good PR too?

The other one from @rany2 has been already merged:
- https://github.com/openwrt/openwrt/pull/19965
